tests/functional/test_direct_edit.py

```diff
@@ -210,6 +210,77 @@ def extract_edit_info(_):
         assert file.is_file()
 
 
+def test_office_temporary_files(manager_factory, obj_factory):
+    """
+    Software from the office suite may leave temporary file behind them.
+    This test goal is to ensure that these files are not removed at startup
+    nor uploaded to the remote by the _cleanup method.
+    Uploading a temporary file may result in data loss on the remote.
+    """
+    manager, engine = manager_factory()
+
+    with manager:
+        direct_edit = manager.direct_edit
+        direct_edit._folder.mkdir()
+        doc = obj_factory(title="test_xlsx.xlsx", nature="File")
+
+        # Creating local excel file
+        folder = direct_edit._folder / f"{doc.uid}_file-content"
+        folder.mkdir()
+        xlsx_file = folder / "test_xlsx.xlsx"
+        xlsx_file.write_text("bla" * 3, encoding="utf-8")
+        assert xlsx_file.is_file()
+
+        # Setting nxdirecteditname attribute on folder
+        direct_edit.local.set_remote_id(
+            folder, "test_xlsx.xlsx", name="nxdirecteditname"
+        )
+
+        # Creating Excel classic temp file
+        tmp_file = folder / "~$test_xlsx.xlsx"
+        tmp_file.write_text("bla" * 3, encoding="utf-8")
+        assert tmp_file.is_file()
+
+        # Creating Excel extensionless temp file
+        random_file = folder / "532AEF06"
+        random_file.write_text("bla" * 3, encoding="utf-8")
+        assert random_file.is_file()
+
+        def extract_edit_info(ref):
+            """
+            Mocked __extract_edit_info() method
+            Mandatory as there is no blob to work with in the test
+            """
+            DirectEditDetails = namedtuple(
+                "details",
+                ["uid", "engine", "digest_func", "digest", "xpath", "editing"],
+            )
+
+            nonlocal doc
+            nonlocal engine
+
+            assert ref == Path(f"{doc.uid}_file-content/{xlsx_file.name}")
+
+            return DirectEditDetails(
+                uid=doc.uid,
+                engine=engine,
+                digest_func="md5",
+                digest="1a36591bceec49c832079e270d7e8b73",
+                xpath="mock",
+                editing="1",
+            )
+
+        with patch.object(
+            direct_edit, "_extract_edit_info", new=extract_edit_info
+        ), ensure_no_exception():
+            direct_edit._cleanup()
+
+        # The files should still be present as they are not deleted by the clean-up
+        assert xlsx_file.is_file()
+        assert tmp_file.is_file()
+        assert random_file.is_file()
+
```

Comment:
Also check the upload queue or whatever object to ensure there will be nothing uploaded.